examples/hpc-c/hello-newlib-hpc/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+cmake_minimum_required(VERSION 3.10)
+
+# Set riscv target
+set(CMAKE_SYSTEM_NAME Generic)
+set(CMAKE_SYSTEM_PROCESSOR riscv64)
+
+project(hello-newlib-hpc C)
+
+# Add C-runtime 0
+set(CRT0 ${CMAKE_CURRENT_SOURCE_DIR})
+
+list(APPEND CMAKE_PREFIX_PATH "/opt/headsail-newlib" "~/work/newlib-build/build")
+
+if(NOT NEWLIB_PATH)
+  find_path(
+    NEWLIB_PATH
+    NAMES "include/" "lib/"
+    PATH_SUFFIXES "build/riscv64-unknown-elf" "riscv64-unknown-elf" "cygwin-newlib/riscv64-unknown-elf"
+  )  
+  if(NOT NEWLIB_PATH)
+    message(FATAL_ERROR "Newlib not found")
+  endif()
+endif()
+
+set(INC_PATH "${NEWLIB_PATH}/include")
+set(LIB_PATH "${NEWLIB_PATH}/lib")
+
+# set(INC_PATH "/opt/headsail-newlib/cygwin-newlib/riscv64-unknown-elf/include")
+# set(LIB_PATH "/opt/headsail-newlib/cygwin-newlib/riscv64-unknown-elf/lib")
```

Comment:
I think this part might belong to a Find-script, e.g., `../cmake/FindNewlib.cmake`. This allows it to be used across the other examples. The variable names likely need to be conventionalized, e.g.,

- `NEWLIB_INCLUDE_DIR := "${NEWLIB_PATH}/include"`
- `NEWLIB_LIBRARIES :="${NEWLIB_PATH}/lib"`.

Reply:
Then include the script like below, after `list(APPEND CMAKE_MODULE_PATH, ...)`: `find_package(Newlib REQUIRED)`.